protocol/communities/community.go

```diff
@@ -1315,20 +1316,64 @@ func (o *Community) Categories() map[string]*protobuf.CommunityCategory {
 	return response
 }
 
-func (o *Community) TokenPermissions() map[string]*protobuf.CommunityTokenPermission {
-	return o.config.CommunityDescription.TokenPermissions
+func (o *Community) tokenPermissions() map[string]*CommunityTokenPermission {
+	result := make(map[string]*CommunityTokenPermission, len(o.config.CommunityDescription.TokenPermissions))
+	for _, tokenPermission := range o.config.CommunityDescription.TokenPermissions {
+		result[tokenPermission.Id] = NewCommunityTokenPermission(tokenPermission)
+	}
+
+	// Non-privileged members should not see pending permissions
```

Comment:
I think this is wrong and tokenPermissions must return all data it has related to the tokenPermissions and not modify the content to return

If permission in a pending state, it must be ignored by privileged members during CommunityEvent handling, wait until Control node handle this event and publish CommunityDescription with the update

Reply:
https://github.com/status-im/status-go/pull/3925#discussion_r1302011133

Reply:
ok, but you already skip it for non-control nodes
https://github.com/status-im/status-go/pull/3925/files#diff-ad79863baafd661e9f1626e2e7ebc0e5db3489ce93c5d5859237446a801b2af2R239-R245

Reply:
Permissions are now obtained also from events, this part is skipping only `CommunityDescription` update (`updateCommunityDescriptionByCommunityEvent`).

Reply:
> ok, but you already skip it for non-control nodes https://github.com/status-im/status-go/pull/3925/files#diff-ad79863baafd661e9f1626e2e7ebc0e5db3489ce93c5d5859237446a801b2af2R239-R245

Permissions are now obtained also from events, this part is skipping only `CommunityDescription` update (`updateCommunityDescriptionByCommunityEvent`).

Reply:
Permissions were obtained from events before, I wrote unit tests for this scenario. All members, who received these permissions, were adding them to their community description. 
From my understanding, your changes should be to ignore from member-side permissions, which are not confirmed by a Control Node.

When you request Community from the database, it should apply all community events on top of the community description. Previously this fn was the same as it was for patching CommunityDescription during handling community events

So you, as a simple member, will not receive patched CommunityDescription with non-approved CommunityPermission. All other admins/token masters must receive it and see it in a pending state